backend/analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/AnalysisScheduler.java

```diff
@@ -64,12 +64,14 @@ public void wakeUp() {
   }
 
   private void executeQueuedCommands() {
-    while (termination.get() == null) {
+    if (termination.get() == null) {
       SonarLintLogger.get().setTarget(logOutput);
+    }
+    while (termination.get() == null) {
       try {
         executingCommand.set(analysisQueue.takeNextCommand());
         if (termination.get() == CANCELING_TERMINATION) {
-          executingCommand.getAndSet(null).cancel();
+          executingCommand.get().cancel();
```

Comment:
In fact, if I'm not wrong, when the termination is set, the executing command is canceled in the `stop` method. So maybe the only thing to do here is to break?

Reply:
No, it's not enough because it will not work on windows. I don't have the specific details what exactly happens in different order. Maybe we should spend time investigating and documenting it because it's not the first time I see discrepancies with windows threading.
But long story short what you written above is only correct for mac/linux.

Reply:
It was already there, so we can leave it like this. But it would be great to understand what we are doing :)

Reply:
For the history nature of the problem:

We start, we wake up executing thread. It start to wait for command. Then we post command from main thread, then we stop scheduler from main thread, executing command atomic was never set yet because executing thread is still sleeping.
So we cleanup the queue during stop, finalize everything. And only then executing thread wakes up and getting new command from the queue. By this moment it sets the command to the Atomic, but stop() already long gone, so there's no one to cancel it.
Then we notice that everything is canceled and break the executing loop.

Solution is to add Thread.sleep(100) to affected tests in order to let scheduler to pickup and properly process the posted command from the queue.